### ISSUE TITLE
Avoid ENOBUFS for large diagnostic output

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1101]
## What to build

Prevent the Node CLI wrapper from failing with `ENOBUFS` when the native linter produces more diagnostic JSON than the default `child_process.spawnSync` output buffer can hold.

With `@utoo/lint@0.2.9`, this public minimal reproduction is sufficient:

```bash
npm init -y
npm install --save-dev @utoo/lint@0.2.9

node -e 'require("node:fs").writeFileSync("fixture.js", "debugger;\n".repeat(20000))'
```

```json
{
  "rules": {
    "no-debugger": "error"
  }
}
```

Save the config as `utlint.config.json`, then run:

```bash
npx utoo-lint --config=utlint.config.json fixture.js --format=json
```

Instead of a JSON report, the wrapper exits with:

```text
spawnSync .../utoo-lint ENOBUFS
```

The native process is currently launched synchronously without an explicit output-buffer strategy, so sufficiently large but valid diagnostic output is lost before it can be parsed or presented to the user.

## Acceptance criteria

- [ ] The CLI can return a valid report containing more than 1 MiB of diagnostics without throwing `ENOBUFS`.
- [ ] Text and JSON output modes both preserve all diagnostics and the correct exit status.
- [ ] The public `run()`/CLI wrapper behavior is covered so programmatic users do not hit the same buffer limit.
- [ ] A regression test generates enough diagnostics to exceed Node's default synchronous child-process buffer.
- [ ] The implementation has an explicit memory strategy, such as streaming output or a documented configurable upper bound, rather than relying on the runtime default.
- [ ] Genuine child-process failures remain distinguishable from an oversized successful lint report.

## Blocked by

None - can start immediately
